pkg/reconciler/clusteringress/resources/secret.go

```diff
@@ -71,6 +72,11 @@ func makeSecret(originSecret *corev1.Secret, targetNamespace string) *corev1.Sec
 				networking.OriginSecretNameLabelKey:      originSecret.Name,
 				networking.OriginSecretNamespaceLabelKey: originSecret.Namespace,
 			},
+			// TODO(zhiminx): currently we temporarily tie the lifecycle of the copied secrets to ClusterIngress.
+			// But this won't work in the future when we have a wildcard certificate because wildecard
+			// certificate can be shared across multiple ClusterIngress. So we need a better way
+			// to handle the lifecyle when landing wildcard certificate.
+			OwnerReferences: []metav1.OwnerReference{*kmeta.NewControllerRef(ci)},
```

Comment:
Could we perhaps do this in a way that we get per-CI copies?

Reply:
I am OK with this. And I made the change accordingly.

My only concern is that we will still make a copy per ClusterIngress when we support single wildcard certificate per namespace, which seems a bit inefficient.

Reply:
Ack :)